src/streamer.ts

```diff
@@ -0,0 +1,36 @@
+import { S3Client } from '@aws-sdk/client-s3';
+import { listBlocks, fetchStreamerMessage } from './s3fetchers';
+import { LakeConfig, BlockHeight, StreamerMessage } from './types';
+
+export async function startStream(
+  config: LakeConfig,
+  onStreamerMessageReceived: (data: StreamerMessage) => Promise<void>
+) {
+  const s3Client = new S3Client({ region: config.s3RegionName });
```

Comment:
We can let user its own `s3Client` and in case it is absent - fallback to our own.

Reply:
Could you elaborate on why this might be useful for the user?

Reply:
@khorolets It allows one to have their own custom S3-compatible client, similarly to https://github.com/near/near-lake-framework/pull/14